dispatches/models/renewables_case/battery.py

```diff
@@ -135,24 +146,95 @@ def build(self):
         self.power_out = Port(noruleinit=True, doc="A port for electricity outflow")
         self.power_out.add(self.elec_out, "electricity")
 
-        @self.Constraint()
-        def state_evolution(b):
-            return b.state_of_charge == b.initial_state_of_charge + (
-                    b.charging_eta * b.dt * b.elec_in
-                    - b.dt / b.discharging_eta * b.elec_out)
-
-        @self.Constraint()
-        def accumulate_energy_throughput(b):
-            return b.energy_throughput == b.initial_energy_throughput + b.dt * (b.elec_in + b.elec_out) / 2
-
-        @self.Constraint()
-        def state_of_charge_bounds(b):
-            return b.state_of_charge <= b.nameplate_energy - b.degradation_rate * b.energy_throughput
-
-        @self.Constraint()
-        def power_bound_in(b):
-            return b.elec_in <= b.nameplate_power
-
-        @self.Constraint()
-        def power_bound_out(b):
-            return b.elec_out <= b.nameplate_power
+        @self.Constraint(self.flowsheet().config.time)
+        def state_evolution(b, t):
+            return b.state_of_charge[t] == b.initial_state_of_charge + (
+                    b.charging_eta * b.dt * b.elec_in[t]
+                    - b.dt / b.discharging_eta * b.elec_out[t])
+
+        @self.Constraint(self.flowsheet().config.time)
+        def accumulate_energy_throughput(b, t):
+            return b.energy_throughput[t] == b.initial_energy_throughput + b.dt * (b.elec_in[t] + b.elec_out[t]) / 2
+
+        @self.Constraint(self.flowsheet().config.time)
+        def state_of_charge_bounds(b, t):
+            return b.state_of_charge[t] <= b.nameplate_energy - b.degradation_rate * b.energy_throughput[t]
+
+        @self.Constraint(self.flowsheet().config.time)
+        def power_bound_in(b, t):
+            return b.elec_in[t] <= b.nameplate_power
+
+        @self.Constraint(self.flowsheet().config.time)
+        def power_bound_out(b, t):
+            return b.elec_out[t] <= b.nameplate_power
+
+    def initialize(self, state_args={}, state_vars_fixed=False,
+                   hold_state=False, outlvl=idaeslog.NOTSET,
+                   temperature_bounds=(260, 616),
+                   solver=None, optarg=None):
+        init_log = idaeslog.getInitLogger(self.name, outlvl, tag="properties")
+        solve_log = idaeslog.getSolveLogger(self.name, outlvl,
+                                            tag="properties")
+        opt = get_solver(solver=solver, options=optarg)
+
+        with idaeslog.solver_log(solve_log, idaeslog.DEBUG) as slc:
+            res = solve_indexed_blocks(opt, [self], tee=slc.tee)
```

Comment:
Could you explain why you are using `solve_indexed_blocks` instead of `opt.solve(self, tee=slc.tee)`? If I understand it correctly, the entire model is in a single block, so the latter should work.

Reply:
fixed!